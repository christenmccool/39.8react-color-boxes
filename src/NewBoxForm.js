import React, {useState} from 'react';
import { Button, InputLabel, TextField } from '@mui/material';
import './NewBoxForm.css';


const NewBoxForm = ({addBox}) => {
  const initialState = {
    color: '#000000',
    width: "200",
    height: "200"
  }
  const [formData, setFormData] = useState(initialState);

  const handleChange = (e) => {
    const {name, value} = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }))
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    addBox(formData);
    setFormData(initialState);
  }


  return (
    <form className="NewBoxForm" >

      <label htmlFor="color">Color</label>
      <input 
        type="color" 
        id="color"
        name="color"
        value={formData.color}
        onChange={handleChange}
      />

      {/* <TextField 
        variant="outlined" 
        label="Height"
        defaultValue="200"
        type="number" 
        name="height"
        value={formData.height}
        onChange={handleChange}
        sx={{ m: 2 }}
      />

      <TextField 
        variant="outlined" 
        label="Width"
        defaultValue="200"
        type="number" 
        name="width"
        value={formData.width}
        onChange={handleChange}
        sx={{ m: 2 }}
      /> */}

      <label htmlFor="height">Height</label>
      <input 
        type="number" 
        id="height"
        name="height"
        value={formData.height}
        onChange={handleChange}
      />

      <label htmlFor="width">Width</label>
      <input 
        type="number" 
        id="width"
        name="width"
        value={formData.width}
        onChange={handleChange}
      />

      <Button variant="contained" onClick={handleSubmit} sx={{ m: 2 }}>Add box</Button>
    </form>
  )
}

export default NewBoxForm;