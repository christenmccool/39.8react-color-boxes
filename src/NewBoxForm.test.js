import { render, screen } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

test('if renders without crashing', () => {
  render(<NewBoxForm />);
});

test("if matches snapshot", () => {
  const {asFragment} = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
})


