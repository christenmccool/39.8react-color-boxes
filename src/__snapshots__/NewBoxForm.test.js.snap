// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewBoxForm"
  >
    <label
      for="color"
    >
      Color
    </label>
    <input
      id="color"
      name="color"
      type="color"
      value="#000000"
    />
    <label
      for="height"
    >
      Height
    </label>
    <input
      id="height"
      name="height"
      type="number"
      value="200"
    />
    <label
      for="width"
    >
      Width
    </label>
    <input
      id="width"
      name="width"
      type="number"
      value="200"
    />
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-s3awaj-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Add box
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </form>
</DocumentFragment>
`;
